### DISCUSSION TITLE
release multiple packages instead of a single one

### DISCUSSION THREAD

**Discussion by ivanhofer** [ivanhofer/typesafe-i18n#173] [Ideas]
Currently the full functionality of `typesafe-i18n` is provided via a single npm package.
So when you install the `typesafe-i18n` package, you get everything at once, which also menas you will download code that you probably will never use.

In version `3.1.0` a migration to `pnpm workspaces` was done. This layed the groundwork of supporting such a scenario.
Packages that would be offered:
 - `@typesafe-i18n/lib`: the core to create `L`, `LL` and `LLL` instances
 - `@typesafe-i18n/angular`: adapter for angular projects
 - `@typesafe-i18n/react`: adapter for `react`-based projects
 - `@typesafe-i18n/svelte`: adapter for `svelte`-based projects
 - `@typesafe-i18n/vue`: adapter for `vue`-based projects
 - `@typesafe-i18n/parser`: `typesafe-i18n`-syntax-parser, others could follow (https://github.com/ivanhofer/typesafe-i18n/discussions/143)
 - `@typesafe-i18n/generator`: package that creates typesafe boilerplate code on top of `@typesafe-i18n/lib` and all `adapters`
 - `@typesafe-i18n/importer`: functionality to import translations from a service
 - `@typesafe-i18n/exporter`: functionality to export translations to a service
 - `@typesafe-i18n/cli`: the programm that detects your setup and runs the `generator`

By splitting the package into multiple packages, we could benefit from:
 - you don't download code you don't need
 - the Docs would split out into multiple READMEs
    each package could contain their specific setup instructions, which would be to much for the main `README`
 - other projects could use e.g. the genrerator to build their own typesafe wrapper (see [`generateonlytypes`-option](https://github.com/ivanhofer/typesafe-i18n#generateonlytypes))
 - would make it more clear that ít supports a specific framework because it's name is included in the package
 
 Downsides of this approach:
 - you need to install multiple packages if you need more functionality
    (the `CLI` should still auto-detect the setup and install e.g. `@typesafe-i18n/svelte` for you)

**Comment by whalemare** [comment]
You also can support behavior like that:

`typesafe-i18n` - "batteries included" solution with all dependencies.
`@typesafe-i18n/lib` - specific library, with only required code

Like [tamagui](https://tamagui.dev/docs/intro/introduction) does with `tamagui` and `@tamagui/core` separation

**Reply by ivanhofer** [reply]
Thanks for sharing the link. This is probably the approach I want to tackle!
